pkg/controllers/routing/network_routes_controller.go

```diff
@@ -342,6 +342,9 @@ func (nrc *NetworkRoutingController) watchBgpUpdates() {
 }
 
 func (nrc *NetworkRoutingController) advertisePodRoute() error {
+	if nrc.MetricsEnabled {
```

Comment:
For a future PR, I think we can make some changes to remove all `if metrics enabled` checks.